api/v1beta2/authpolicy_types.go

```diff
@@ -161,13 +184,13 @@ type AuthPolicySpec struct {
 
 	// The auth rules of the policy.
 	// See Authorino's AuthConfig CRD for more details.
-	AuthScheme AuthSchemeSpec `json:"rules,omitempty"`
+	AuthScheme *AuthSchemeSpec `json:"rules,omitempty"`
```

Comment:
Switched to pointer so that this will not show on json `spec` as empty object `{}` when empty which would be a bit confusing to a user, especially if `spec.defaults` is used instead 🤔 Also makes the mutual exclusive CEL rule a bit easier by switching to pointer